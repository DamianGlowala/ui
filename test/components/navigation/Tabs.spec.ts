import { describe, it, expect } from 'vitest'
import { UTabs } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'
import uiTabs from '../../../src/runtime/ui.config/navigation/tabs'

describe('Tabs', () => {
  it.each([
    ['basic case', { props: { ui: uiTabs } }],
    ['with orientation', { props: { orientation: 'vertical', ui: uiTabs } }],
    ['with items', { props: { items: [{ label: 'Item 1' }, { label: 'Item 2' }], ui: uiTabs } }],
    ['with tabs slot', { slots: { default: 'tab slot' }, props: { items: [{ label: 'Item 1' }, { label: 'Item 2' }], ui: uiTabs } }]
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof Tabs.props>) => {
    const html = await ComponentRender(nameOrHtml, options, UTabs)
    expect(html).toMatchSnapshot()
  })
})
