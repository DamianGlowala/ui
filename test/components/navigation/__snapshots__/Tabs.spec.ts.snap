// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tabs > renders basic case correctly 1`] = `
"<button aria-hidden="true" style="position: fixed; height: 0px; padding: 0px; overflow: hidden; clip: rect(0, 0, 0, 0); white-space: nowrap; border-width: 0px;" type="button"></button>
<div>
  <div role="tablist" aria-orientation="horizontal" class="inline-grid items-center" style="grid-template-columns: repeat(0, minmax(0, 1fr));">
    <div class="">
      <div class="w-full h-full"></div>
    </div>
  </div>
  <div class=""></div>
</div>"
`;

exports[`Tabs > renders with items correctly 1`] = `
"<div>
  <div role="tablist" aria-orientation="horizontal" class="inline-grid items-center" style="grid-template-columns: repeat(2, minmax(0, 1fr));">
    <div class="" style="top: 0px; left: 0px; width: 0px; height: 0px;">
      <div class="w-full h-full"></div>
    </div><button class="relative inline-flex items-center justify-center flex-shrink-0 w-full ui-focus-visible:outline-0 ui-focus-visible:ring-2 ui-focus-visible:ring-primary-500 dark:ui-focus-visible:ring-primary-400 ui-not-focus-visible:outline-none focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 transition-colors duration-200 ease-out" id="headlessui-tabs-tab-0" role="tab" aria-selected="true" tabindex="0" data-headlessui-state="selected" type="button" aria-controls="headlessui-tabs-panel-2"><span class="truncate">Item 1</span></button><button class="relative inline-flex items-center justify-center flex-shrink-0 w-full ui-focus-visible:outline-0 ui-focus-visible:ring-2 ui-focus-visible:ring-primary-500 dark:ui-focus-visible:ring-primary-400 ui-not-focus-visible:outline-none focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 transition-colors duration-200 ease-out" id="headlessui-tabs-tab-1" role="tab" aria-selected="false" data-headlessui-state="" type="button" aria-controls="headlessui-tabs-panel-3"><span class="truncate">Item 2</span></button>
  </div>
  <div class="">
    <div id="headlessui-tabs-panel-2" role="tabpanel" tabindex="0" data-headlessui-state="selected" class="" aria-labelledby="headlessui-tabs-tab-0"></div>
    <div id="headlessui-tabs-panel-3" role="tabpanel" hidden="" style="display: none;" data-headlessui-state="" class="" aria-labelledby="headlessui-tabs-tab-1"></div>
  </div>
</div>"
`;

exports[`Tabs > renders with orientation correctly 1`] = `
"<button aria-hidden="true" style="position: fixed; height: 0px; padding: 0px; overflow: hidden; clip: rect(0, 0, 0, 0); white-space: nowrap; border-width: 0px;" type="button"></button>
<div>
  <div role="tablist" aria-orientation="vertical" class="">
    <div class="">
      <div class="w-full h-full"></div>
    </div>
  </div>
  <div class=""></div>
</div>"
`;

exports[`Tabs > renders with tabs slot correctly 1`] = `
"<div>
  <div role="tablist" aria-orientation="horizontal" class="inline-grid items-center" style="grid-template-columns: repeat(2, minmax(0, 1fr));">
    <div class="" style="top: 0px; left: 0px; width: 0px; height: 0px;">
      <div class="w-full h-full"></div>
    </div><button class="relative inline-flex items-center justify-center flex-shrink-0 w-full ui-focus-visible:outline-0 ui-focus-visible:ring-2 ui-focus-visible:ring-primary-500 dark:ui-focus-visible:ring-primary-400 ui-not-focus-visible:outline-none focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 transition-colors duration-200 ease-out" id="headlessui-tabs-tab-4" role="tab" aria-selected="true" tabindex="0" data-headlessui-state="selected" type="button" aria-controls="headlessui-tabs-panel-6">tab slot</button><button class="relative inline-flex items-center justify-center flex-shrink-0 w-full ui-focus-visible:outline-0 ui-focus-visible:ring-2 ui-focus-visible:ring-primary-500 dark:ui-focus-visible:ring-primary-400 ui-not-focus-visible:outline-none focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 transition-colors duration-200 ease-out" id="headlessui-tabs-tab-5" role="tab" aria-selected="false" data-headlessui-state="" type="button" aria-controls="headlessui-tabs-panel-7">tab slot</button>
  </div>
  <div class="">
    <div id="headlessui-tabs-panel-6" role="tabpanel" tabindex="0" data-headlessui-state="selected" class="" aria-labelledby="headlessui-tabs-tab-4"></div>
    <div id="headlessui-tabs-panel-7" role="tabpanel" hidden="" style="display: none;" data-headlessui-state="" class="" aria-labelledby="headlessui-tabs-tab-5"></div>
  </div>
</div>"
`;
